### PR TITLE
Implement MS16257: Remove 'clear overlay when moving' setting and code

### PR DESCRIPTION
**Pull request by zfox23**
Implements [MS16257](https://highfidelity.fogbugz.com/f/cases/16257/Remove-the-setting-Settings-General-Clear-Overlays-when-moving).

This fixes recent frustrations our users of Master builds have had because of the implementation of PR #13341. According to @alexiamandeville:
>We see no value, and users are frustrated by it. The option to hide them will persist.

### CODE REVIEW

**Comment by zfox23** [comment]
## Test Plan
1. Starting from a clean install of this build, verify that Overlays (the HUD, etc) are visible in Desktop mode by default. Also verify that, in the View menu, "Show Overlays" is `checked`.
2. In Desktop mode, use your movement/fly keys to drive/fly around. Verify that the HUD stays visible.
3. Under the View menu, `uncheck` "Show Overlays". Verify that your overlays (HUD, etc) disappear.
4. Under the View menu, `check` "Show Overlays". Verify that your overlays (HUD, etc) re-appear.
5. Don your HMD. Open the "HMD Friendly script log window". Move your head around and verify that the script log window remains visible.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10946/HighFidelity-Beta-PR13485-cfc8c0f.apk

File: interface/src/ui/OverlayConductor.cpp [hunk 1]
```diff
@@ -115,7 +104,7 @@ void OverlayConductor::update(float dt) {
         }
     }
 
-    bool targetVisible = Menu::getInstance()->isOptionChecked(MenuOption::Overlays) && (0 == (_flags & SuppressMask));
+    bool targetVisible = Menu::getInstance()->isOptionChecked(MenuOption::Overlays) && (0 == (_flags & SuppressedByHead));
```
Comment (danteruiz) [inline interface/src/ui/OverlayConductor.cpp]: If you are not moving does unchecking `show overlays` work? Nissim uses that for is automated testing

Comment (zfox23) [reply interface/src/ui/OverlayConductor.cpp]: Yes - video sent via Slack.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10950/HighFidelity-Beta-PR13485-2347634.apk

File: interface/src/ui/OverlayConductor.h [hunk 2]
```diff
@@ -26,12 +26,10 @@ class OverlayConductor {
...
 
     enum SupressionFlags {
-        SuppressedByMove = 0x01,
-        SuppressedByHead = 0x02,
-        SuppressMask = 0x03,
+        SuppressedByHead = 0x01
     };
 
-    uint8_t _flags { SuppressedByMove };
+    uint8_t _flags { SuppressedByHead };
```
Comment (danteruiz) [inline interface/src/ui/OverlayConductor.h]: Do you think we are going to add more flags down the line? If not we just make this in a bool.

Comment (zfox23) [reply interface/src/ui/OverlayConductor.h]: I have no idea :) But making it a bool for now seems like a good idea.

Comment (danteruiz) [reply interface/src/ui/OverlayConductor.h]: Lets do that.

File: interface/src/ui/OverlayConductor.h [hunk 3]
```diff
@@ -29,7 +29,7 @@ class OverlayConductor {
         SuppressedByHead = 0x01
```
Comment (danteruiz) [inline interface/src/ui/OverlayConductor.h]: remove the enum

File: interface/src/ui/OverlayConductor.cpp [hunk 4]
```diff
@@ -93,22 +93,22 @@ void OverlayConductor::update(float dt) {
 
     bool shouldRecenter = false;
 
-    if (_flags & SuppressedByHead) {
+    if (_suppressedByHead) {
         if (isAtRest) {
```
Comment (danteruiz) [inline interface/src/ui/OverlayConductor.cpp]: isAtReset could be defined inside this if and also isMoving is not used.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10952/HighFidelity-Beta-PR13485-8ba184d.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10955/HighFidelity-Beta-PR13485-1b414fb.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10955/HighFidelity-Beta-PR13485-1b414fb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10955/HighFidelity-Beta-Interface-PR13485-1b414fb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10955/HighFidelity-Beta-PR13485-1b414fb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10955/HighFidelity-Beta-Interface-PR13485-1b414fb.exe

**Comment by qliejahner** [comment]
PASS @conklin94122 

No more vanishing menus!

**Pull request merged**